add KMM samples

hjanuschka opened the pull request:
sample on how we use a shared fastfile across multiple apps with and without app-extensions

Commit:
add KMM samples

hjanuschka commented:
had to remove app-id's and git urls from internal servers/apps - sorry

hjanuschka commented:
added our android lanes to the sample 
note: the android lanes need the adb/adb_device/gradle(modified) actions -> see this PR: https://github.com/fastlane/fastlane/pull/1130

KrauseFx commented on KroneMultimedia/SharedFastfile.rb:
```diff
@@ -0,0 +1,289 @@
+
+  kill_simulator
+  snapshot( {
+      workspace: $self_workspace,
```
Seems like the tabs are wrong here. You can use `rubocop -a` and point to this file to fix it :+1:

KrauseFx commented on KroneMultimedia/SharedFastfile.rb:
```diff
@@ -0,0 +1,289 @@
+   before_build
+   certificates
+
+  gym({
```
Could you remove all the `{` in the Ruby configuration files and just have `( ... )`?

KrauseFx commented on KroneMultimedia/SharedFastfile.rb:
```diff
@@ -0,0 +1,289 @@
+  version = app.edit_version
+
+  
+    #get latest build
```
Wrong spacing here too, should be fixed with `rubocop -a` :rocket:

KrauseFx commented on KroneMultimedia/SharedFastfile.rb:
```diff
@@ -0,0 +1,289 @@
+   submission.export_compliance_is_exempt = false
+   submission.export_compliance_uses_encryption = false
+   submission.export_compliance_app_type = nil
+   submission.export_compliance_encryption_updated = false
```
This should actually be part of `deliver`, but I guess you had troubles using `deliver` to submit your app for Review? Is this lane just to submit an existing build for Review, right? 
What do you think about adding this functionality to `deliver`?

KrauseFx commented:
Hey @hjanuschka, thank you so much for this PR. It looks great, however there are some style improvements before it's ready to be merged. See the comments on how you can fix those automatically.

hjanuschka commented:
ok i rebased to a single commit, did a quick rubocop, fixed almost all issues (only longlines, global vars, are left)

about the commet that `:submit_build` lane should be inside deliver. i totally agree you that this should be in deliver, maybe i am doing a PR there too.

we usually have the workflow:
- create app version on iTC
- invite testflight testers
- submit a new TF build on every git commit - auto increment build nr.
- invite external testers
- resign ipa and invite crowdtesting platforms (won't do this on every app/version)
- once we get a "clear" sign from the internal QA (wich coordinates the testers)
- we may have even newer builds already deployed to testflight, but the "clear" build should go into release, therefore we needed the submit_build
- once the build is submitted and the review is done we start again with a new version train.

hjanuschka commented:
update: did the PR on `deliver` -> https://github.com/fastlane/deliver/pull/548

mfurtak commented:
@KrauseFx I don't think I am in touch with how you would like examples to be, so I will let you see this through!

KrauseFx commented:
Hey @hjanuschka, thanks for the pull request, however we're working on the mono repo right now and won't merge this before the switch. Let's re-connect about this in a few weeks :+1:

hjanuschka commented:
@KrauseFx  i would love to reopen that one - until now we even added `sharethemeal` and `aws` testing - wich i would provide with a further PR - whats your opinion on that?

we are using it for following apps:
- krone news (ios & android)
- krone sport (ios & android)
- willi (ios & android, www.willi.at)

and a few inhouse apps, with the policy -> no app without shared fastlane!
